sdk/resource/config.go

```diff
@@ -67,6 +55,8 @@ func (d detectAttributes) Detect(context.Context) (*Resource, error) {
 }
 
 // WithDetectors adds detectors to be evaluated for the configured resource.
+// If no detectors are passed, a resource will default to use BuiltinDetectors.
+// To disable this behavior use a NoOp Detector
```

Comment:
```suggestion
// If no detectors are passed, construct the resource using BuiltinDetectors.
// To disable this behavior, use a NoOp Detector.
```
This still seems odd. If one skips using `WithDetectors`, what happens? Does the provider use `BuiltinDetectors`? The comment at line 81 below suggests that it does. If one decides to use `WithDetectors`, but with an empty sequence, it seems that that should mean something different than not having used `WithDetectors` at all. I interpret that instinctively to mean "with no `Detector`s." Restating:
- `New(ctx)`: Use builtin `Detector`s.
- `New(ctx, WithDetectors())`: Use no `Detector`s.
- `New(ctx, WithDetectors(d1, d2))`: Use `Detector` "d1," then overlay `Detector` "d2"'s findings.

Another approach to consider is changing the signature of `WithDetectors` to mandate at least one `Detector`: `WithDetectors(first Detector, rest ...Detector)`. However, we're then back to the problem of being able to indicate that we want no `Detectors`, which could lead to another function like `WithoutDetectors()`. If someone passes both `WithDetectors` and `WithoutDetectors`, we could either say that the latter would have no effect, or we could attempt to catch the conflict and return an error from `New`.

Reply:
That is a good point.  I updated `WithDetectors` to handle that case, it should now disable the defaults if you pass no Detectors.   

I think that approach will put us effectively in the same position.  We would also have to handle the special case when someone uses `WithDetectors(nil)`.